pkg/daemon/daemon.go

```diff
@@ -196,10 +191,11 @@ func (dn *Daemon) Run(stop <-chan struct{}) error {
 
 	// Catch quickly if we've been asked to run once.
 	if dn.onceFrom != "" {
-		if dn.fromIgnition {
+		switch configType := SenseOnceFromConfigType(dn.onceFrom); configType {
+		case MachineConfigIgnitionFileType:
 			glog.V(2).Info("Daemon running directly from Ignition")
-			return dn.runFromIgnition()
-		} else {
+			return dn.runOnceFromIgnition()
+		default:
 			glog.V(2).Info("Daemon running once per request")
 			return dn.runOnce()
```

Comment:
I'd suggest renaming this function to runOnceFromMachineConfig (if that is indeed the case) just for naming clarity/consistency. Otherwise, I love this restructure!

Reply:
Updated

Reply:
Yesss looks good!